cmake/profile.cmake

```diff
@@ -1,4 +1,30 @@
 # Copyright (c) 2020 ARM Limited. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
-include(${MBED_ROOT}/cmake/profiles/${MBED_PROFILE}.cmake)
+# Set Develop as our default type if no other selected 
+if(NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE "Develop" CACHE
+        STRING "The type of build" FORCE)
+endif()
+
+# Mapping CMAKE_BUILD_TYPE into MBED_BUILD_TYPES, as we understand only 3 profiles
+set(MBED_BUILD_TYPES Debug Release Develop)
+
+# Force the build types to be case-insensitive for checking
+set(LOWERCASE_MBED_BUILD_TYPES ${MBED_BUILD_TYPES})
+list(TRANSFORM LOWERCASE_MBED_BUILD_TYPES TOLOWER)
+string(TOLOWER ${CMAKE_BUILD_TYPE} LOWERCASE_CMAKE_BUILD_TYPE)
+
+get_property(multi_config GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+if(multi_config)
+    # multi configuration generators do not support build type, providing a list only
+    set(CMAKE_CONFIGURATION_TYPES "${MBED_BUILD_TYPES}" CACHE STRING "List of supported build types" FORCE)
+else()
+    # Set the possible values of build type for cmake-gui
+    set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "${MBED_BUILD_TYPES}")
+    if(NOT LOWERCASE_CMAKE_BUILD_TYPE IN_LIST LOWERCASE_MBED_BUILD_TYPES)
```

Comment:
Maybe a empty line between the code at line 24 and 25 so it is clear that the comment only applies to line 24.